File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumber.java
package com.example.webmvc.favouriteNumber;

import com.example.webmvc.registration.User;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class FavouriteNumber {
    @Id
    @GeneratedValue
    private Long id;

    private int number = 0;

    @ManyToOne
    private User user;

    public FavouriteNumber(int number, Long id) {
        this.number = number;
        this.id = id;
    }

    public FavouriteNumber(int number) {
        this.number = number;
    }

    public FavouriteNumber() {
    }

    public FavouriteNumber(FavouriteNumber number) {
        this(number.getNumber(), number.getId());
        this.setUser(number.getUser());
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumber.kt
package com.example.webmvc.favouriteNumber

import com.example.webmvc.favouriteNumber.FavouriteNumber
import com.example.webmvc.registration.User
import org.hibernate.Hibernate
import javax.persistence.*

@Entity
data class FavouriteNumber(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var number: Int,
) {
    @ManyToOne
    var user: User? = null

    constructor(number: Int) : this(null, number)
    constructor() : this(null, 0)
    constructor(number: FavouriteNumber) : this(number.id, number.number)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as FavouriteNumber

        return id != null && id == other.id
    }

    override fun hashCode(): Int = 0

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , number = $number , user = $user )"
    }
}